File: fizzbuzz/fizzbuzz.py
"""

    * A number is fizz if it is divisible by 3 or if it has a 3 in it
    * A number is buzz if it is divisible by 5 or if it has a 5 in it

    1
    2
    Fizz
    4
    Buzz
    Fizz
    7
    8
    Fizz
    Buzz
    11
    Fizz
    13
    14
    FizzBuzz
    16
    17
    Fizz
    19
    Buzz

"""

def fizzbuzz(number):
    if number % 5 == 0 or "{}".format(number)[-1] == "5":
        if number % 3 == 0:
            return "Fizzbuzz"
        else:
            return "Buzz"
    elif number % 3 == 0 or "{}".format(number)[-1] == "3":
        return "Fizz"
    return number



if __name__ == "__main__":
    assert(fizzbuzz(1) == 1)
    assert(fizzbuzz(2) == 2)
    assert(fizzbuzz(3) == "Fizz")
    assert(fizzbuzz(4) == 4)
    assert(fizzbuzz(5) == "Buzz")
    assert(fizzbuzz(6) == "Fizz")
    assert(fizzbuzz(7) == 7)
    assert(fizzbuzz(8) == 8)
    assert(fizzbuzz(9) == "Fizz")
    assert(fizzbuzz(10) == "Buzz")
    assert(fizzbuzz(15) == "Fizzbuzz")
    print("All Test passed!")
    print("-" * 32)

File: fizzbuzz/fizzbuzz.js
const assert = require("assert");

const fizzbuzz = (number) => {
    if (number % 5 == 0 || number.toString(10).substr(-1) == "5") {
        if (number % 3 == 0) {
            return "Fizzbuzz";
        } else {
            return "Buzz";
        }
    } else if (number % 3 == 0 || number.toString(10).substr(-1) == "3") {
        return "Fizz";
    }
    return number;
};

describe("Simple Math Test", () => {
    it("should return 1", () => {
        assert.equal(fizzbuzz(1), 1);
    });
    it("should return 2", () => {
        assert.equal(fizzbuzz(2), 2);
    });
    it("should return Fizz", () => {
        assert.equal(fizzbuzz(3), "Fizz");
    });
    it("should return 4", () => {
        assert.equal(fizzbuzz(4), 4);
    });
    it("should return Buzz", () => {
        assert.equal(fizzbuzz(5), "Buzz");
    });
    it("should return 6", () => {
        assert.equal(fizzbuzz(6), "Fizz");
    });
    it("should return 7", () => {
        assert.equal(fizzbuzz(7), 7);
    });
    it("should return 8", () => {
        assert.equal(fizzbuzz(8), 8);
    });
    it("should return Fizz", () => {
        assert.equal(fizzbuzz(9), "Fizz");
    });
    it("should return 10", () => {
        assert.equal(fizzbuzz(10), "Buzz");
    });
    it("should return 11", () => {
        assert.equal(fizzbuzz(11), 11);
    });
    it("should return 12", () => {
        assert.equal(fizzbuzz(12), "Fizz");
    });
    it("should return 13", () => {
        assert.equal(fizzbuzz(13), "Fizz");
    });
    it("should return 14", () => {
        assert.equal(fizzbuzz(14), 14);
    });
    it("should return 15", () => {
        assert.equal(fizzbuzz(15), "Fizzbuzz");
    });
});
